### PR TITLE
ca4x: Add accurate frequency data, comment on A coef sources

### PR DESCRIPTION
**Pull request by pathfinder49**
Add accurate transition frequencies with citation #35

Edit: notably the new ca43 frequencies include the isotope shift

### CODE REVIEW

**Comment by hartytp** [comment]
@pathfinder49 have you checked that these are the most up to date numbers? This should be a relatively simple thing to do: look at the table in DJS's thesis, which gives references from ~10 years ago. Most numbers won't have newer data than that, but it's easy to check by searching for things that cite the papers that DJS cites. A quick skim over the titles should give you the answer.

I'd prefer to hold off updating this until someone has done that basic due diligence.

**Comment by pathfinder49** [comment]
The only new data are the transition frequencies. Therese are all sourced from this 2020 publication: https://doi.org/10.1016/j.adt.2019.101322. Feel free to do any other checks. However, I would hope such checks were done in peer review. The paper claims 

> All available experimental data on isotope shifts and absolute frequency measurements in the optical spectra of Ca I and Ca II are re-evaluated

**Comment by pathfinder49** [comment]
The table from the DJS thesis is below. For ca40, the dipole transitions agree within the error quoted of the 10 year old measurements.

729 transition disagrees outide of error. The value in this commit is based on more precise measurements from 2019:
Knollmann F.W., Patel A.N., Doret S.C. Phys. Rev. A, 100 (2019), Article 022514

Regarding the isotope shift:
The numbers agree within error, except the 854 transition. both the DJS theis and the summary paper use the same underlying data. However the summary paper notes significant systematic error which were corrected.

> Collinear laser spectroscopy of fast ion beams was also used by Nörtershäuser et al. [16] to measure the IS in the Ca II 3d–4p transitions (3–4, 3–4, and 3–4) for isotopes 42, 43, 44, 46, and 48 (relative to 40Ca). Correction of significant systematic errors present in the results of Ref. [16] is described in Appendix A.3.


image

**Comment by pathfinder49** [comment]
I assume this is now fine to merge.

File: ion_phys/ions/ca40.py [hunk 1]
```diff
@@ -25,55 +25,55 @@ def __init__(self, *, B=None, level_filter=None):
             None we include all levels.
         """
         levels = {
-            ground_level: LevelData(g_J=2.00225664),
-            P12: LevelData(),
-            P32: LevelData(),
-            D32: LevelData(),
-            D52: LevelData()
+            ground_level: LevelData(g_J=2.00225664),  # [?]
```
Comment (hartytp) [inline ion_phys/ions/ca40.py:28]: this one shouldn't be a question mark. You can cite my thesis until we publish the g factor measurement properly

File: ion_phys/ions/ca40.py [hunk 2]
```diff
@@ -25,55 +25,55 @@ def __init__(self, *, B=None, level_filter=None):
...
+            P12: LevelData(),  # [?]
+            P32: LevelData(),  # [?]
```
Comment (hartytp) [inline ion_phys/ions/ca40.py:30]: why add a question mark here?

**Comment by hartytp** [comment]
thanks! that's a useful contribution, much appreciated. I haven't checked the numbers or uncertainties (nor the original source of the data that's tabulated there), but it's clearly a thorough paper from NIST and all looks sensible.

I had a couple of nits about some of the question marks, but other than that please do merge!

**Review by hartytp — APPROVED** [review]

**Pull request merged**